### ISSUE TITLE
Create a hyperparameter scan plotting app

### ISSUE THREAD

**Issue by scarlehoff** [NNPDF/nnpdf#828]
Right now the hyperscan plotting script is really just a script with a series of option that have been added to it regularly as they were needed.
Now that the methodology has converged it would be necessary to create a true plotting app that generates a report with all the different options.

The bare minimum for such a script is to read all the `tries.json` found in the replica folders, merge it and print out a table with the best hyperparameters like:

```
All trials:
  optimizer  learning_rate  clipnorm  epochs  stopping_patience  stopping_epochs  ...        avg  std               hlosses               vlosses iteration layer_5
5   RMSprop       0.012632  1.194554   420.0               0.08             33.6  ...  31.743887  0.0  [31.743886910232842]  [0.6463762273643222]         5     NaN
6   RMSprop       0.065268  1.771893   300.0               0.16             48.0  ...  51.195920  0.0   [51.19592045802696]   [3.274243504867941]         6     NaN

[2 rows x 27 columns]
Best setup:
  optimizer learning_rate clipnorm epochs stopping_patience stopping_epochs  \
5   RMSprop     0.0126316  1.19455    420              0.08            33.6   

  pos_initial pos_multiplier       nodes_per_layer number_of_layers  \
5        None           None  [32.0, 8.0, 24.0, 8]                3   

  activation_per_layer layer_type layer_1 layer_2 layer_3 layer_4  \
5              sigmoid      dense      32       8      24     NaN   

      initializer dropout  good validation_loss     loss      avg std  \
5  glorot_uniform    0.15  True        0.646376  31.7439  31.7439   0   

                hlosses               vlosses iteration layer_5  
5  [31.743886910232842]  [0.6463762273643222]         5     NaN
```

Other goodies the script should have are:
- For each hyperparameter, a scatter plot (or violin plot for discrete options) with the selected values.
- A hierarchical organization. i.e., the optimizer Adam and Adadelta will both have learning rates, but they will be independent of each other.
- A filter, so that one can set for instance `--filter "number_of_layers==4"` to select only such trials (this is the only thing that is already well implemented -more or less-)
- The possibility of setting a custom loss: while the default should be taking the hyperopt loss as the loss, we have all the information of the fit in the `json` dictionary so we can also generate reports for the "best worst loss", or the best training, or whatever.
